### ISSUE TITLE
False warning: pattern matching on a field doesn't count as a use

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#52956]
```cs
class C
{
    C field1 = null;
    public static void Main()
    {        
        _ = new C() is { field1: {} }; // CS0414 (not expected)
    }
}
```
```cs
class C
{
    C field1;
    public static void Main()
    {        
        _ = new C() is { field1: {} }; // CS0169 (not expected)
    }
}
```

Test plan: https://github.com/dotnet/roslyn/issues/52468

**Comment by Youssef1313** [comment]
I expect there is one or more `NoteRead` calls are missing in https://github.com/dotnet/roslyn/blob/6a28cdcfdaf3770db0b360e72398813dc7483de8/src/Compilers/CSharp/Portable/FlowAnalysis/DefiniteAssignment.cs#L1591

**Comment by jcouv** [comment]
Thanks for finding this. I'm fixing it as part of the extended property patterns.

**Comment by jcouv** [comment]
Fixed by https://github.com/dotnet/roslyn/pull/53594 in feature branch (planned to merge for 17.0)
